docs/sources/setup-grafana/configure-security/configure-authentication/okta/index.md

```diff
@@ -1,166 +1,146 @@
+## Before you begin
 
-## Create an Okta application
+To follow this guide:
 
-Before you can sign a user in, you need to create an Okta application from the Okta Developer Console.
+- Ensure that you have access to the [Grafana configuration file]({{< relref "../../../configure-grafana#configuration-file-location" >}}).
+- Ensure you know how to create an OAuth2 application with your OAuth2 provider. Consult the documentation of your OAuth2 provider for more information.
+- If you are using refresh tokens, ensure you know how to set them up with your OAuth2 provider. Consult the documentation of your OAuth2 provider for more information.
 
-1. Log in to the [Okta portal](https://login.okta.com/).
+## Integrate Okta OAuth2 with Grafana
 
-1. Go to Admin and then select **Developer Console**.
+To integrate your Okta OAuth2 provicer with Grafana using our Okta OAuth2 integration, follow these steps:
 
-1. Select **Applications**, then **Add Application**.
+1. [Create SWA app integrations](https://help.okta.com/en-us/Content/Topics/Apps/Apps_App_Integration_Wizard_SWA.htm) section.
 
-1. Pick **Web** as the platform.
+1. Set the callback URL for your OAuth2 app to `http://<my_grafana_server_name_or_ip>:<grafana_server_port>/login/generic_oauth`.
 
-1. Enter a name for your application (or leave the default value).
+   Ensure that the callback URL is the complete HTTP address that you use to access Grafana via your browser, but with the appended path of `/login/generic_oauth`.
 
-1. Add the **Base URI** of your application, such as https://grafana.example.com.
+   For the callback URL to be correct, it might be necessary to set the root_url option to [server], for example, if you are serving Grafana behind a proxy.
 
-1. Enter values for the **Login redirect URI**. Use **Base URI** and append it with `/login/okta`, for example: https://grafana.example.com/login/okta.
+1. Refer to the following table to update field values located in the `[auth.generic_oauth]` section of the Grafana configuration file:
 
-1. Click **Done** to finish creating the Okta application.
+   | Field                        | Description                                                                                                   |
+   | ---------------------------- | ------------------------------------------------------------------------------------------------------------- |
+   | `client_id`, `client_secret` | These values must match the client ID and client secret from your Okta OAuth2 app.                            |
+   | `auth_url`                   | The authorization endpoint of your OAuth2 provider. `https://<okta-tenant-id>.okta.com/oauth2v1authorize`     |
+   | `token_url`                  | The token endpoint of your Okta OAuth2 provider. `https://<okta-tenant-id>.okta.com/token`                    |
+   | `api_url`                    | The user information endpoint of your Okta OAuth2 provider. `https://<tenant-id>.okta.com/oauth2/v1/userinfo` |
```

Comment:
I think we can skip these from the table, as the default value will work for them.